include/wx/prntbase.h

```diff
@@ -284,8 +284,11 @@ class WXDLLIMPEXP_CORE wxPrintout: public wxObject
 
     virtual bool HasPage(int page);
     virtual bool OnPrintPage(int page) = 0;
+
     virtual void GetPageInfo(int *minPage, int *maxPage, int *pageFrom, int *pageTo);
-    virtual bool IsPageSelected(int page);
+
+    // the default implementation of this new method uses the older GetPageInfo method above
+    virtual void GetPageInfoRanges(int* minPage, int* maxPage, std::vector<wxPrintPageRange>* pageRanges);
```

Comment:
I think we should return the ranges from here instead of using an output pointer, this would be cleaner:
```suggestion
    virtual void std::vector<wxPrintPageRange> GetPageInfoRanges(int* minPage, int* maxPage);
```